### PR TITLE
Update extension API validation

### PR DESCRIPTION
**Pull request by RedworkDE**
Followup to https://github.com/godotengine/godot/pull/76647#issuecomment-1554519390, originally just trying to ensure that ignoring the hash change of one method does not hide any future changes to that method, but has somewhat grown in scope to tracking all changes that possibly interfere with compatibility, not just those that break gdextension binary compatibility.

Check `misc/extension_api_validation/4.0-stable.expected` for how the new errors look like currently.

Basically the only changes to existing APIs that it now doesn't try to complain about are changing parameter names and increasing the value of an enums `_MAX` value.

### CODE REVIEW

**Comment by dsnopek** [comment]
I didn't test this, but skimming the code and looking at the changes to the .expected file, this looks awesome!

**Review by akien-mga — APPROVED** [review]
Looks good to me.

**Comment by YuriSizov** [comment]
Thanks!
